external-resolver/src/main/scala/org/genivi/sota/resolver/DeviceDataMigrator.scala

```diff
@@ -0,0 +1,43 @@
+package org.genivi.sota.resolver
+
+import org.genivi.sota.common.DeviceRegistry
+import org.genivi.sota.resolver.db.ForeignPackages
+import org.genivi.sota.resolver.db.ForeignPackages.InstalledForeignPackage
+import org.genivi.sota.marshalling.CirceMarshallingSupport._
+import org.slf4j.LoggerFactory
+import io.circe.syntax._
+import io.circe.generic.auto._
+import scala.concurrent.duration._
+import slick.driver.MySQLDriver.api._
+
+import scala.concurrent.{Await, ExecutionContext, Future}
+
+object DeviceDataMigrator {
+
+  private lazy val logger = LoggerFactory.getLogger(this.getClass)
+
+  def apply(deviceRegistry: DeviceRegistry)(implicit db: Database, ec: ExecutionContext): Unit = {
+
+    val resultF = db.run(ForeignPackages.foreignPackages.result).map { rows =>
```

Comment:
if this works, for now it's fine.

In the future you might want to look at `db.stream`. Then you can add a `Sink` on that stream to process each row individually and contact device registry. This way you get automagic backpressure support and error handling, without even needing to put the whole table in memory. You also won't need to wait on that future since it will just run until done.